service/controller/resource/clustersecret/desired.go

```diff
@@ -81,7 +86,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		},
 	}
 
-	if !reflect.ValueOf(r.proxy).IsZero() {
+	if privateCluster && !reflect.ValueOf(r.proxy).IsZero() {
```

Comment:
`privateCluster` is always `false` for `infra.VCDClusterKind` meaning that it won't go thru this if. Is that expected?

Reply:
good catch!!!

Reply:
fixed

Reply:
it'd be the same for `VSphereClusterKind`; which is not covered in the case above.